modules/xmpp/xmpp.js

```diff
@@ -498,6 +498,15 @@ export default class XMPP extends Listenable {
                 const { features, identities } = this.caps.parseDiscoInfo(msg);
 
                 this._processDiscoInfoIdentities(identities, features);
+
+                // check for shard name in identities
+                if (identities) {
```

Comment:
I think this test is redundant (even an empty Set evaluates to `true`), but I don't know if there's a way to express that in js.

Reply:
Ups, I will drop it.

Reply:
Done.